src/sharebuttonview.js

```diff
@@ -50,6 +51,12 @@ class ShareButtonView {
       + "/sheet/" + SheetID + "/state/analysis/options/clearselections";
   }
 
+  encodeString(str) {
+    return encodeURIComponent(str).replace(replaceRegEx, (c) => {
+      return `%${c.charCodeAt(0).toString(16).toUpperCase()}`;
+    });
+  }
```

Comment:
I don't know if this is the correct encoding to use
(or if all uses of id should use the same encoding)

(consider changing the use of id so that the escaping is not needed)

Reply:
I can have a look and check if we can only use the encodeURIComponent method which is already being used in the extension for app id.

Reply:
using encodeURIComponent for this feels wrong as the id is not used as an url parameter but as an attribute on a html element.
(it may work correct in most or all cases but is not the type of encoding encodeURIComponent is intended for)

Reply:
I was also thinking the same and that's the reason I used encodeString method which I took reference from sense-client, but may be then I shall see if the usage of id can be changed in a way that we avoid encoding.

Reply:
I have used encoding still, but now the way id is used has been changed, so with the encoded id, the xss issue exists no more and the user can use the object as expected.

Reply:
consider updating
https://github.com/qlik-oss/Dashboard-Link-Generator/blob/f9fb8691ca1c71efcb54d1e966c6ae14aa4b5551/src/sharebuttonview.js#L275
that is the use if id with the largest risk